### ISSUE TITLE
Unable to access Sharepoint with Client ID and Secret

### ISSUE THREAD

**Issue by ManikandharV** [SharePoint/sp-dev-docs#9378]
### What type of issue is this?

Question

### What SharePoint development model, framework, SDK or API is this about?

SharePoint CSOM

### Target SharePoint environment

SharePoint Online

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [X] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

.net core console application

### Issue description

We have a background job, which runs continuously to make some changes on SharePoint List.
We are creating SharePoint context using "GetACSAppOnlyContext".

But we are getting random errors from over "ExecuteQuery" method. Below are the list of errors we are getting.
[Documentation followed](https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azureacs#using-this-principal-in-your-application-using-the-pnp-framework-library)

![image](https://github.com/SharePoint/sp-dev-docs/assets/153075845/99d423cf-5ff2-469e-ac7f-71ca4126e2cc)

These error are happening randomly.
Please let me know how to solve them as soon as possible.

**Comment by waaromikniet** [comment]
Have you tried the sample code? Or are you using custom code? Maybe post your code and config (without secret).

**Comment by ManikandharV** [comment]
I am not using custom code, I am following Microsoft documentation.
Below is the code reference that I used to build.
[Documentation](https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azureacs#using-this-principal-in-your-application-using-the-pnp-framework-library)

![image](https://github.com/SharePoint/sp-dev-docs/assets/153075845/becfccb7-4e84-4152-b80f-1fb34751f972)

![image](https://github.com/SharePoint/sp-dev-docs/assets/153075845/c6ae602a-f6d3-43b1-ba9b-d8f02db68925)

Error occurs randomly in the code, sometimes it occurs at the start when creating context, and some times it happens in between the code. This started happening since November 27th.

Is it happening because Microsoft is retiring ACS based access?

**Comment by waaromikniet** [comment]
I don't think it's because of retirement of ACS. I use it a lot and am not facing this issue. Seems a lot like this issue mentioned in PnP.Framework. 

https://github.com/pnp/pnpframework/issues/292

**Comment by waaromikniet** [comment]
Do you have this setting enabled?

https://learn.microsoft.com/en-us/answers/questions/866093/drawbacks-and-disadvantageous-of-setting-disablecu

**Comment by ManikandharV** [comment]
@waaromikniet ,

Yes I check all the parameters, and I disabled custom app authentication. Yet the issue persists.

![image](https://github.com/SharePoint/sp-dev-docs/assets/153075845/b9e7e3b8-007b-48b6-a61e-89d8a45245e1)

**Comment by waaromikniet** [comment]
You are connecting to a list. Can you try getting the title of the web? 

Can you provide the permission scope you added using appinv.aspx?

**Comment by ManikandharV** [comment]
No, I am getting Token request failed with inner exception of SSL forcibly closed error.

Permissions:
`<AppPermissionRequests AllowAppOnlyPolicy="true">
<AppPermissionRequest Scope="http://sharepoint/content/tenant" Right="FullControl" />
</AppPermissionRequests>`

**Comment by ManikandharV** [comment]
> pnp/pnpframework#292

Seems like this issue is not resolved even in that thread you mentioned.

**Comment by waaromikniet** [comment]
Have you tested the clientid/secret using Postman for example? Try connecting with Postman and validate that that works. It could be a TLS setting on your device running the console app. 

See

https://techcommunity.microsoft.com/t5/microsoft-sharepoint-blog/provider-hosted-app-pha-application-errors-tls-errors-and-401/ba-p/2273611

**Comment by ManikandharV** [comment]
Yes, I tried, TLS too... nothing working. That's why I raised this issue.

**Comment by waaromikniet** [comment]
Did you try creating a different sitecollection and a new app registration?
Maybe not granting tenant permissions but sitecollection and see what happens?
When I follow the example I don't get the error.
